### PR TITLE
Digest auth

### LINKED ISSUE #12: Digest authentication support

### PR DESCRIPTION
**Pull request by ghostbuster91**
This closes #12

### CODE REVIEW

File: core/shared/src/main/scala/sttp/client/RequestT.scala [hunk 1]
```diff
@@ -320,6 +320,11 @@ class SpecifyAuthScheme[U[_], T, +S](hn: String, rt: RequestT[U, T, S]) {
 
   def bearer(token: String): RequestT[U, T, S] =
     rt.header(hn, s"Bearer $token")
+
+  def digest(user: String, password: String): RequestT[U, T, S] = {
+    import DigestAuthenticationBackend._
+    rt.digestAuth(user, password)
```
Comment (adamw) [inline core/shared/src/main/scala/sttp/client/RequestT.scala]: shouldn't this take into account `hn` - the header name? (whether it's proxy or not)

**Comment by adamw** [comment]
We're just missing the docs on how to use digest authentication :)

File: core/shared/src/main/scala/sttp/client/RequestT.scala (hunk 1 above)
Comment (ghostbuster91) [reply core/shared/src/main/scala/sttp/client/RequestT.scala]: I don't think so, it just sets a tag on request which is later read by `DigestAuthenticatorBackend`

Comment (adamw) [reply core/shared/src/main/scala/sttp/client/RequestT.scala]: but shouldn't `request.auth.digest` work differently from `request.proxyAuth.digest`?

Comment (ghostbuster91) [reply core/shared/src/main/scala/sttp/client/RequestT.scala]: The proxy path works a little bit different than normal path, but it is all in `DigestAuthenticationBackend`

**Pull request merged**